src/utils/runtime/src/lib.rs

```diff
@@ -33,7 +36,23 @@ use prof::*;
 
 /// Start RisingWave components with configs from environment variable.
 ///
-/// Currently, the following env variables will be read:
+/// # Shutdown on Ctrl-C
+///
+/// The given closure `f` will take a [`CancellationToken`] as an argument. When a `SIGINT` signal
+/// is received (typically by pressing `Ctrl-C`), [`CancellationToken::cancel`] will be called to
+/// notify all subscribers to shutdown. You can use [`.cancelled()`](CancellationToken::cancelled)
+/// to get notified on this.
+///
+/// Users may also send a second `SIGINT` signal to force shutdown. In this case, this function
+/// won't return and the process will exit with code 1.
+///
+/// When `f` returns, this function will assume that the component has finished its work and it's
+/// safe to exit. Therefore, the main runtime to drive the future will be shutdown **without**
+/// waiting for background tasks to finish. Then, the output of `f` will be returned.
```

Comment:
It seems the output of `f` is always useless?

Reply:
You're right. We can also enforce `f` to return unit type.

Reply:
+1. A good entrypoint function should indeed return nothing